src/Controller/Admin/RealmSigningLogCrudController.php

```diff
@@ -0,0 +1,134 @@
+    {
+        return $crud
+            ->setPageTitle('index', 'Users');
+    }
+
+    public function configureFields(string $pageName): iterable
+    {
+        return [
+            IdField::new('serial'),
+            TextField::new('requester'),
+            TextField::new('subjectWithoutCustomerName')
+                ->setLabel('Pseudo account')
+                ->hideOnIndex(),
+            TextField::new('realm'),
+            TextField::new('client'),
+            DateTimeField::new('issued')
+                ->setFormat('yyyy-MM-dd HH:mm:ss')
+                ->formatValue(function ($value, $entity) {
+                    return $value ? $value : '-';
+                }),
+            DateTimeField::new('revoked')
+                ->setFormat('yyyy-MM-dd HH:mm:ss')
+                ->formatValue(function ($value, $entity) {
+                    return $value ? $value : '-';
+                })
+        ];
+    }
+
+    public function configureActions(Actions $actions): Actions
+    {
+        $revoke = Action::new('revoke', 'Revoke')
+                ->linkToCrudAction('revokeRealmSigningLog')
+                ->addCssClass('confirm-action')
+                ->setHtmlAttributes([
+                    'data-bs-toggle' => 'modal',
+                    'data-bs-target' => '#modal-confirm',
+                ]);
+
+        return parent::configureActions($actions)
+            ->disable(Action::EDIT)
+            ->disable(Action::NEW)
+            ->disable(Action::DELETE)
+            ->add(Crud::PAGE_INDEX, $revoke)
+            ->add(Crud::PAGE_INDEX, Action::DETAIL)
+            ->addBatchAction(Action::new('revoke batch', 'Revoke batch')
+            ->linkToCrudAction('revokeRealmSigningLogBatch')
+            ->addCssClass('btn btn-primary')
+            ->setIcon('fa fa-user-check'));
+
+    }
+
+    public function configureAssets(Assets $assets): Assets
+    {
+        $assets->addJsFile('assets/js/confirm-modal.js');
+
+        return parent::configureAssets($assets);
+    }
+
+    public function configureFilters(Filters $filters): Filters
+    {
+        return parent::configureFilters($filters)
+            ->add(TextFilter::new('realm'))
+            ->add(DateTimeFilter::new('revoked'));
+    }
+
+    /*
+     * LinkToCrudAction to revoke one realm signing log, configured at configureActions
+     */
+    public function revokeRealmSigningLog(AdminContext $context) : Response
+    {
+        try {
+            $entity = $context->getEntity()->getInstance();
+            $entityManager = $this->container->get('doctrine')->getManagerForClass(RealmSigningLog::class);
+            $repository = $entityManager->getRepository(RealmSigningLog::class);
+            $repository->revoke($entity, true);
+        } catch (ForeignKeyConstraintViolationException $e) {
+            throw new EntityRemoveException(['entity_name' => $context->getEntity()->getName(), 'message' => $e->getMessage()]);
+        }
```

Comment:
Wellicht wat netter om het hele riedeltje hierboven naar een Service achtige class te verhuizen.

Reply:
Deze zal ik meenemen in de volgende PR.

Reply:
Prima, en dan logischerwijs ook de hieronderstaande method meenemen.